lib/ansible/plugins/loader.py

```diff
@@ -296,6 +383,37 @@ def _find_plugin(self, name, mod_type='', ignore_deprecated=False, check_aliases
             # they can have any suffix
             suffix = ''
 
+        # TODO: we should still probably require a well-known prefix on FQNS so we don't paint ourselves into a corner with future stuff
+        if ('.' in name or collection_list) and not name.startswith('Ansible'):  # HACK: need this right now so we can still load shipped PS module_utils
+            if '.' in name or not collection_list:
+                candidates = [name]
+            else:
+                candidates = ['{0}.{1}'.format(c, name) for c in collection_list]
+            # TODO: keep actual errors, not just assembled messages
+            errors = []
+            for candidate_name in candidates:
+                try:
+                    # HACK: refactor this properly
+                    if candidate_name.startswith('ansible.legacy'):
+                        # just pass the raw name to the old lookup function to check in all the usual locations
+                        p = self._find_plugin_legacy(name.replace('ansible.legacy.', '', 1), ignore_deprecated, check_aliases, suffix)
+                    else:
+                        p = self._find_fq_plugin(candidate_name, suffix)
+                    if p:
+                        return p
+                except Exception as ex:
```

Comment:
Catching exceptions wholesale is bad. Can we narrow it down to something more specific?

Reply:
Potentially a nasty game of whack-a-mole if I don't; I've seen about 5 different exceptions just from the import side across various Python versions (we're subject to whatever exception any Python loader we might encounter feels like throwing, /me glares at `ZipLoader`)... We could improve the logging to serialize tracebacks, but if we let an unexpected exception fly, it could potentially break existing stuff on the legacy path due to one of those "who cares" failures in the collections path.